weld/codegen/llvm2/mod.rs

```diff
@@ -1245,8 +1259,60 @@ impl LlvmGenerator {
                     unreachable!()
                 }
             }
-            Sort { .. } => {
-                unimplemented!() // Sort
+            Sort { ref child, ref cmpfunc } => { // cmpfunc is an SirFunction
+                let output_pointer = context.get_value(output)?;
+                let output_type = context.sir_function.symbol_type(
+                    statement.output.as_ref().unwrap())?;
+                
+                if let Vector(ref elem_ty) = *output_type {
+                    let child_value = self.load(context.builder, context.get_value(child)?)?;
+                    let child_type = context.sir_function.symbol_type(child)?;
+
+                    use ast::Type::Scalar;
+                    use ast::ScalarKind;
+                    use self::vector::VectorExt;
+                    let zero = self.zero(self.i64_type());
+                    let child_elems = self.gen_at(context.builder, child_type, child_value, zero)?;
+                    let elems_ptr = LLVMBuildBitCast(context.builder, child_elems,
+                                                     self.void_pointer_type(),
+                                                     c_str!(""));
+                    let size = self.gen_size(context.builder, child_type, child_value)?;
+                    let mut elem_ll_ty = self.llvm_type(&*elem_ty)?;
+                    let ty_size = self.size_of(elem_ll_ty);
+                    
+                    use self::cmp::GenCmp;
+                    let cmpfunc_ll_fn = self.functions[cmpfunc];
+
+                    let run = context.get_run();
+
+                    // Generate the comparator from the provided custom code.
+                    // The custom comparator can only make use of BinOps that are supported by Weld
```

Comment:
Maybe delete this comment? Since you can also have `If`, `GetField`, etc etc

Reply:
The point was you can't use BinOps on dicts or something where there isn't a default comparator, but I see how this comment is confusing